applications/minotari_node/src/stratum/tari_sha3x_stratum_handler.rs

```diff
@@ -0,0 +1,356 @@
+        if job.is_none() {
+            return Err(anyhow::anyhow!("Job with id {} not found", job_id));
+        }
+        let job = job.unwrap();
+
+        // Quick check for the target.
+        let result = hex::decode(result)?;
+        let target = &hex::decode(job.target)?;
+        let result_u64 = if job.pow_algo == PowAlgorithm::RandomXT {
+            let result = &result[result.len() - 8..];
+            dbg!(&result);
+            u64::from_le_bytes([
+                result[0], result[1], result[2], result[3], result[4], result[5], result[6], result[7],
+            ])
+        } else {
+            u64::from_be_bytes([
+                result[0], result[1], result[2], result[3], result[4], result[5], result[6], result[7],
+            ])
+        };
+        let target_u64 = u64::from_le_bytes([
+            target[0], target[1], target[2], target[3], target[4], target[5], target[6], target[7],
+        ]);
+        if result_u64 > target_u64 {
+            return Err(anyhow::anyhow!("Result is greater than target"));
+        }
+
+        let (tx, rx) = tokio::sync::oneshot::channel();
+        let _res = self
+            .submit_job_queue_sender
+            .send((
+                SubmittedJob {
+                    job_id: job.job_id.to_string(),
+                    nonce,
+                    pow_algo: job.pow_algo.clone(),
+                    original_mining_hash: job.original_mining_hash.clone(),
+                    cuckaroo_nonces: cuckaroo_nonces.unwrap_or_default(),
+                },
+                tx,
+            ))
+            .await?;
+        let res = rx.await?;
+        match res {
+            Ok(_) => {
+                // Handle successful submission
+                Ok(SubmitResponse {
+                    id,
+                    result: true, // Placeholder for actual result
+                })
+            },
+            Err(e) => {
+                return Err(anyhow::anyhow!("Failed to submit job: {}", e));
+            },
+        }
+        // Handle submit request
+    }
+
+    async fn subscribe(&self, _id: String, _agent: String) -> anyhow::Result<SubscribeResponse> {
+        let mut r = OsRng;
+        let subscription_id = hex::encode(r.next_u64().to_le_bytes());
+
+        let nonce = rand::random::<u16>();
+
+        let xn = hex::encode(&nonce.to_le_bytes());
+
+        Ok(SubscribeResponse {
+            subscription_id: subscription_id.clone(),
+            nonce_hex: xn.clone(),
+            nonce,
+        })
+    }
+
+    async fn authorize(
+        &self,
+        _id: String,
+        main_algo: String,
+        login: String,
+        _worker_name: Option<String>,
+        _pass: String,
+        nonce: Option<u16>,
+    ) -> anyhow::Result<AuthorizeResponse> {
+        let algo = main_algo.parse()?;
+        let address = TariAddress::from_str(&login)?;
+
+        let job_record = create_job_from_blob(&self.block_template_repository, algo, address, nonce).await?;
```

Comment:
`nonce` is not used in `create_job_from_blob`
```suggestion
        let job_record = create_job_from_blob(&self.block_template_repository, algo, address, nonce).await?;
```